scripts/cluster_providers/kind.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash -e
+
+function create_cluster(){
+   kind create cluster --name $2 --image kindest/node:v$1
+   wait_till_cluster_up $2
+}
+
+function delete_cluster() {
+   kind delete cluster --name $2 
+}
+
+function get_cluster_auth() {
+   echo "kubectl config set-context kind-$2"
```

Comment:
we are not retrieving the auth details here, we are just changing the cluster name here. not even setting the current setting. could you please paste a sample output here?

Reply:
@bamachrn  
output of `get_cluster_auth()` essentially returns cluster auth information by printing them. 
1. For kind related providers this would be returning string command which would set cluster context before making call to k8s api. which would be `kubectl config set-context kind-cluster-name`.
2. For other providers where we would get kubeconfig file path, auth command would be `export KUBECONFIG=path`

Therefore when we execute any k8s api using `helm` or `kubectl`  call would be following
1. _Kind -_ ` kubectl config set-context kind-cluster-name && kubectl get pods`
2. _Kubeconfig-_ `export KUBECONFIG=path && kubectl get pods`

Reply:
in essence we dont need a kubeconfig file for login to the `kind` cluster.  Then where are we adding the cluster, user and authentication details in kubeconfig? I mean in order to kubectl set the context there should be an existing context created, or we can update the current context. does this makes sense @akashshinde ?

Reply:
@bamachrn kind appends kubeconfig information to default config path i.e $HOME/.kube/config